src/wp-includes/media.php

```diff
@@ -987,9 +984,111 @@ function wp_get_attachment_image_src( $attachment_id, $size = 'thumbnail', $icon
 	 * @param int          $attachment_id Image attachment ID.
 	 * @param string|int[] $size          Requested image size. Can be any registered image size name, or
 	 *                                    an array of width and height values in pixels (in that order).
-	 * @param bool         $icon          Whether the image should be treated as an icon.
 	 */
-	return apply_filters( 'wp_get_attachment_image_src', $image, $attachment_id, $size, $icon );
+	return apply_filters( 'wp_get_attachment_preview_src', $image, $attachment_id, $size );
+}
+
+/**
+ * Retrieves an image to represent an attachment.
+ *
+ * @since 2.5.0
+ *
+ * @param int          		$attachment_id Image attachment ID.
+ * @param string|int[] 		$size          Optional. Image size. Accepts any registered image size name, or an array of
+ *                                    	   width and height values in pixels (in that order). Default 'thumbnail'.
+ * @param bool|string|array $args          {
```

Comment:
I'm not sure we need to support a string here, that would only be for mime_type right? I think its fine to ask developers to use array ( 'mime_type' => ...) instead of supporting a string.  That makes the API more flexible for the future.

Reply:
I tend to agree now we should only keep the short-hand form allowed for `$icon`, since that is needed for backward compatibility. This also makes it safer since it would allow us to interpret anything which `! is_array` as value for `$icon`. People may be (somewhat incorrectly, but so far not problematic) a non-empty or empty string to this function for `$icon`, and we shouldn't now suddenly change that to be interpreted as a MIME type.

So +1 on only going with `bool|array` for documented parameters, and thus only supporting the old `$icon` way to provide this parameter for backward compatibility.